client/cache/bloom_filter_cache_test.go

```diff
@@ -0,0 +1,139 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/bits-and-blooms/bloom/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+type Mock struct {
+	Db      map[string]any
+	loadCnt int64
+}
+
+func TestBloomFilterCache(t *testing.T) {
```

Comment:
refactor the test to be table driven, like:
```go
testCases = []struct {
    name string
    mock func() // can return something or accept something
    clean func()
    wantVal any
    wantErr error
}
```
So the test cases are independent and more readable